src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -69,26 +69,8 @@ std::tuple<gl_sframe, gl_sframe> drawing_classifier::init_data(
       gl_sframe data, variant_type validation_data) const {
   gl_sframe train_data;
   gl_sframe val_data;
-  if (variant_is<gl_sframe>(validation_data)) {
-    train_data = data;
-    val_data = variant_get_value<gl_sframe>(validation_data);
-    /* TODO: Do we want to log_and_throw when the validation data is empty?
-     *       Having it will trigger a unit test failure!
-     */
-  }
-  else if ((variant_is<flex_string>(validation_data)) 
-      && (variant_get_value<flex_string>(validation_data)=="auto")) {
-    size_t num_rows = data.size();
-    if (num_rows >= 100) {
-      std::tie(train_data, val_data) = data.random_split(.95);
-    } else {
-      train_data = data;
-      std::cout << "The dataset has less than the minimum of 100 rows required for train-validation split. "
-                       "Continuing without validation set.\n";
-    }
-  } else {
-    train_data = data;
-  }
+  std::tie(train_data, val_data) = turi::supervised::create_validation_data(
+      data, validation_data);
```

Comment:
Wouldn't it be more clear just to have this function be a one-liner? Just
```
return turi::supervised::create_validation_data(data, validation_data);
```

Reply:
```
std::tie(train_data, val_data) =
      turi::supervised::create_validation_data(data, validation_data);
```
google style.

Reply:
`create_validation_data` returns a `std::pair`, whereas we want to return a `std::tuple`. While `std::pair` is a special case of `std::tuple`, would directly returning `turi::supervised::create_validation_data(data, validation_data)` not throw an error?

Since we want to merge this PR asap, relevant internal tests have passed, and @Jarvi-Izana is depending on it, let's merge this now and get style changes come in later PRs.

Reply:
No, it would not throw an error. `std::pair<T,U>` should be implicitly convertible to `std::tuple<T,U>`

Reply:
sounds good to me